tests/integration/store-test.js

```diff
@@ -202,12 +203,16 @@ test("destroying the store correctly cleans everything up", function(assert) {
   assert.equal(person.get('cars.firstObject'), car, " expected persons cars's firstRecord to be the correct car");
 
   run(store, 'destroy');
-
   assert.equal(personWillDestroy.called.length, 1, 'expected person to have recieved willDestroy once');
   assert.equal(carWillDestroy.called.length, 1, 'expected car to recieve willDestroy once');
   assert.equal(carsWillDestroy.called.length, 1, 'expected person.cars to recieve willDestroy once');
   assert.equal(adapterPopulatedPeopleWillDestroy.called.length, 1, 'expected adapterPopulatedPeople to recieve willDestroy once');
   assert.equal(filterdPeopleWillDestroy.called.length, 1, 'expected filterdPeople.willDestroy to have been called once');
+  if (config.environment !== 'production') {
+    assert.throws(() => {
```

Comment:
I think we generally use `assert.expectAssertion(() => {}, /some matcher/)` for this.  That custom assertion is provided by https://github.com/workmanw/ember-qunit-assert-helpers/ and is aware of development vs production...

Reply:
@rwjblue - If I do that, then the production test still fails, because it moves on and hits the original `TypeError: Cannot read property 'push' of null`.

This is why I had originally built the assertion into a conditional.  It would allow an early return - in production, the assertion would be skipped, but you wouldn't hit that error.

Reply:
Or another idea:  Maybe we want to just `throw` instead of `assert`?  i.e. maybe we _want_ this better error message in production too? This case is currently throwing an error in prod either way.

Reply:
@rwjblue ⬆️  Any thoughts on this?

Reply:
Pinging again... @rwjblue ⬆️   Any thoughts?

Reply:
Thanks @runspired !  Any thoughts on the other issues above?  If we want to leave production alone, we'll have to think of some way to only call that method in the non-prod tests.

Reply:
@danwenzel see https://github.com/emberjs/data/blob/master/tests/unit/store/push-test.js#L7

Reply:
Thanks, @runspired , that worked perfectly!  All passing now